src/feature/featuremodel.cpp

```diff
@@ -157,144 +157,116 @@ QObject* FeatureModel::get(const QString& feature) {
+    if (!experimentalFeature) {
+      logger.warning() << "Got" << key
+                       << "but experimental feature doesn't exist. Ignoring.";
+      experimentalFeaturesToToggleOn.removeAll(key);
       continue;
     }
-    AdjustFiltering::instance()->allowField(param.toString());
-  }
 
-  QJsonValue denyParameterValue = adjustFieldsValue["denyParameters"];
-  if (!denyParameterValue.isObject()) {
-    logger.error()
-        << "Error in the json format; deny parameters in not an object";
-    return;
-  }
-
-  QJsonObject denyParameterObject = denyParameterValue.toObject();
-  for (const QString& key : denyParameterObject.keys()) {
-    QJsonValue value = denyParameterObject.value(key);
-    if (!value.isString()) {
-      logger.error()
-          << "Error in the json format; deny list parameter is not a string";
-      continue;
+    QJsonValue experimentalFeatureSettings = experimentalFeaturesObj[key];
+    if (!experimentalFeatureSettings.isObject()) {
+      logger.error() << "Error in the json format: experimentalFeature" << key
+                     << "is not an object.";
+      experimentalFeaturesToToggleOn.removeAll(key);
+      return;
     }
 
-    AdjustFiltering::instance()->denyField(key, value.toString());
-  }
+    QJsonObject experimentalFeatureSettingsObj =
+        experimentalFeatureSettings.toObject();
+    for (const QString& settingKey : experimentalFeatureSettingsObj.keys()) {
+      auto value = experimentalFeatureSettingsObj[settingKey].toVariant();
+      if (!value.isValid() || value.isNull()) {
+        logger.warning()
+            << "Received null value for experimental feature setting"
+            << settingKey;
+        continue;
+      }
 
-  QJsonValue mirrorParameterValue = adjustFieldsValue["mirrorParameters"];
-  if (!mirrorParameterValue.isObject()) {
-    logger.error()
-        << "Error in the json format; mirror parameters are not an object";
-    return;
-  }
+      logger.debug() << "Setting experimental feature setting" << settingKey;
 
-  QJsonObject mirrorParameterObject = mirrorParameterValue.toObject();
-  for (const QString& key : mirrorParameterObject.keys()) {
-    QJsonValue values = mirrorParameterObject.value(key);
-    if (!values.isArray()) {
-      logger.error() << "Error in the json format; mirror parameters value is "
-                        "not an array";
-      continue;
+      experimentalFeature->settingGroup()->set(settingKey, value);
     }
+  }
 
-    QJsonArray valuesArray = values.toArray();
-    if (valuesArray.size() != 2) {
-      logger.error()
-          << "Error in the json format; mirror value is not an array";
-      continue;
-    }
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
+  // Experimental features that are not in the paylod must be disabled, so
+  // overwriting the whole list here is indeed what we want.
+  settingsHolder->setFeaturesFlippedOn(experimentalFeaturesToToggleOn);
```

Comment:
Won't this overwrite features that are not in the group and which were previously flipped on?

Reply:
Ack. Good catch yet again. You know, when I was implementing this I was thinking "wow, this was easier than I thought...".

Reply:
Yeah well, it's a good thing we are not using the featuresOverwrite thingy, because that is totally broken. I will fix only the experimental features part here and file a follow up for the features overwrite one.

Reply:
https://mozilla-hub.atlassian.net/browse/VPN-6078

Reply:
I fixed this on [4040239](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8904/commits/404023952bef8c745784943edd728eaafc7d0e1d). Will add more comprehensive testing next week.